server/swap/swap.go

```diff
@@ -1221,48 +1240,39 @@ func (s *Swapper) checkInaction(assetID uint32) {
 
 		switch match.Status {
 		case order.NewlyMatched:
-			if match.makerStatus.swapAsset != assetID {
-				return
-			}
-			// If the maker is not acting, the swapTime won't be set. Check against
-			// the time the match notification was sent (match.time) for the broadcast
-			// timeout.
-			if match.makerStatus.swapTime.IsZero() && match.time.Before(oldestAllowed) {
+			// Maker has not broadcast their swap. They have until match time
+			// plus bTimeout.
+			if match.time.Before(oldestAllowed) {
 				failMatch(true)
 			}
 		case order.MakerSwapCast:
-			if match.takerStatus.swapAsset != assetID {
-				return
-			}
-			// If the maker has sent their swap tx, check the taker's broadcast
-			// timeout against the time of the swap's SwapConf'th confirmation.
-			if match.takerStatus.swapTime.IsZero() &&
-				!match.makerStatus.swapConfirmed.IsZero() &&
```

Comment:
Since the status is `MakerSwapCast`, `takerStatus.swapTime` cannot be set.  When it gets set, it would be `TakerSwapCast`.